minicluster/src/main/java/tachyon/master/LocalTachyonCluster.java

```diff
@@ -249,6 +250,17 @@ private void startMaster(TachyonConf testConf) throws IOException {
     // Update the test conf with actual RPC port.
     testConf.set(Constants.MASTER_PORT, String.valueOf(getMasterPort()));
 
+    // If we are using the LocalMiniDFSCluster, we need to update the UNDERFS_ADDRESS to point to
+    // the cluster's current address. This must happen here because the cluster isn't initialized
+    // until mMaster is started.
```

Comment:
I think in the future, a cleaner way to do this is to check the `ufs` system property (or something equivalent set by profile `hdfs1Test` and etc). After detecting this flag is set, in `newTestConf`, we update `UNDERFS_ADDRESS` accordingly. And in `UnderFileSystemCluster `, we could create corresponding ufs according to the conf.